stan/math/prim/mat/prob/ordered_probit_lpmf.hpp

```diff
@@ -0,0 +1,223 @@
+#ifndef STAN_MATH_PRIM_MAT_PROB_ORDERED_PROBIT_LPMF_HPP
+#define STAN_MATH_PRIM_MAT_PROB_ORDERED_PROBIT_LPMF_HPP
+
+#include <stan/math/prim/scal/fun/Phi.hpp>
+#include <stan/math/prim/scal/fun/log1m.hpp>
+#include <stan/math/prim/scal/err/check_bounded.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/scal/err/check_greater.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/arr/err/check_ordered.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <Eigen/StdVector>
+#include <string>
+#include <vector>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Returns the (natural) log probability of the specified integer
+     * outcome given the continuous location and specified cutpoints
+     * in an ordered probit model.
+     *
+     * <p>Typically the continous location
+     * will be the dot product of a vector of regression coefficients
+     * and a vector of predictors for the outcome.
+     *
+     * @tparam propto True if calculating up to a proportion.
+     * @tparam T_loc Location type.
+     * @tparam T_cut Cut-point type.
+     * @param y Outcome.
+     * @param lambda Location.
+     * @param c Positive increasing vector of cutpoints.
+     * @return Log probability of outcome given location and
+     * cutpoints.
+     * @throw std::domain_error If the outcome is not between 1 and
+     * the number of cutpoints plus 2; if the cutpoint vector is
+     * empty; if the cutpoint vector contains a non-positive,
+     * non-finite value; or if the cutpoint vector is not sorted in
+     * ascending order.
+     */
+    template <bool propto, typename T_loc, typename T_cut>
+    typename return_type<T_loc>::type
+    ordered_probit_lpmf(int y, const T_loc& lambda,
+                          const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c) {
+      using std::exp;
+      using std::log;
+
+      static const std::string function = "ordered_probit";
+
+      int K = c.size() + 1;
+
+      check_bounded(function, "Random variable", y, 1, K);
+      check_finite(function, "Location parameter", lambda);
+      check_greater(function, "Size of cut points parameter", c.size(), 0);
+      check_ordered(function, "Cut-points", c);
+      check_finite(function, "Cut-points", c);
+
+      if (y == 1) {
+        return log1m(Phi(lambda - c[0]));
+      } else if (y == K) {
+        return log(Phi(lambda - c[K-2]));
+      } else {
+        return log(Phi(lambda - c[y-2]) - Phi(lambda - c[y-1]));
+      }
+    }
+
+    template <typename T_loc, typename T_cut>
+    typename return_type<T_loc>::type
+    ordered_probit_lpmf(int y, const T_loc& lambda,
+                          const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c)  {
```

Comment:
When possible, try to line continued arguments directly underneath previous line.  So here `int` and `const` should be left aligned on lines 70 and 71.

The reason for all these rules isn't to obsess about order, but to make code easier to scan for people reading it.  Code's a lot harder to read than it is to right, so we try to put the burden on the author to make it readable.